### PR TITLE
Post-mergedown update for DiseasesReimagined, DrywallHidesPipes, and ExpandedLights.

### PR DESCRIPTION
**Pull request by medkg15**
This is a first pass on updating 3 mods for post-mergedown.  Haven't done a ton of testing yet but the mods load and the primary features are working.

Thanks for making these mods!

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -20,11 +20,11 @@
     <ItemGroup>
       <InputAssemblies
         Include="$(TargetDir)\*.dll"
-        Exclude="$(TargetPath)" />
+        Exclude="$(TargetPath);$(TargetDir)\Microsoft.*.dll;$(TargetDir)\System.*.dll;" />
```
Comment (medkg15) [inline Directory.Build.targets:23]: ILRepack was adding other assemblies to SkyLib which caused a ReflectionLoadTypeException crash.

File: DiseasesReimagined/BuildingsPatch.cs [hunk 2]
```diff
@@ -51,7 +52,7 @@ public static void Prefix()
         [HarmonyPatch(typeof(Db),"Initialize")]
         public static class Db_Initialize_Patch
         {
-            public static void Prefix()
+            public static void Postfix()
```
Comment (medkg15) [inline DiseasesReimagined/BuildingsPatch.cs:55]: Adding buildings in prefix was crashing - moved to Postfix.

File: DiseasesReimagined/DiseasesPatch.cs [hunk 3]
```diff
@@ -33,11 +35,15 @@ public static void OnLoad()
...
             ImaginationLoader.Init(typeof(DiseasesPatch));
-            PUtil.RegisterPatchClass(typeof(CompatPatch));
-            PUtil.RegisterPatchClass(typeof(DiseasesPatch));
-            BuildingsPatch.uvlight = PLightShape.Register("SkyLib.LightShape.FixedSemi",
+            var patchManager = new PPatchManager(harmony);
+            patchManager.RegisterPatchClass(typeof(CompatPatch));
+            patchManager.RegisterPatchClass(typeof(DiseasesPatch));
+            BuildingsPatch.uvlight = new PLightManager().Register("SkyLib.LightShape.FixedSemi",
                 BuildingsPatch.SemicircleLight);
+            FrostbitePatch.Mod_OnLoad.OnLoad();
```
Comment (medkg15) [inline DiseasesReimagined/DiseasesPatch.cs:46]: Can only have one UserMod2 per DLL (maybe Mod?) - calling FrostbitePatch from here as a workaround.

File: DiseasesReimagined/ImaginationLoader.cs [hunk 4]
```diff
@@ -73,8 +74,10 @@ public sealed class ImaginationLoader {
 #if DEBUG
             if (loader == null)
                 PUtil.LogWarning("IsFinalDestination() called before save loaded!");
-#endif
-            return loader?.worldGen?.Settings?.world?.name == FINALDEST_KEY;
+#endif 
+            // Not sure what the new API is for this.
+            // return loader?.worldGen?.Settings?.world?.name == FINALDEST_KEY;
+            return false;
```
Comment (medkg15) [inline DiseasesReimagined/ImaginationLoader.cs:80]: Need to figure out the new way to do this, if still needed.

File: DiseasesReimagined/MoreEvilFlower.cs [hunk 5]
```diff
@@ -35,11 +35,14 @@ public void Sim4000ms(float dt)
...
-                    SimMessages.ModifyDiseaseOnCell(cell, disease, Mathf.RoundToInt(
-                        GermExposureTuning.SPORECHID_GERMS_PER_S * dt));
+                foreach (var monitoredCell in uprooted.monitorCells)
+                {
+                    cell = Grid.OffsetCell(cell, monitoredCell);
+                    if (Grid.IsValidCell(cell) && Grid.Solid[cell])
+                        // Flower is growing and on a solid cell, infect it!
+                        SimMessages.ModifyDiseaseOnCell(cell, disease, Mathf.RoundToInt(
+                            GermExposureTuning.SPORECHID_GERMS_PER_S * dt));
+                }
```
Comment (medkg15) [inline DiseasesReimagined/MoreEvilFlower.cs:45]: I think this is the correct thing to do - API changed from a single cell to an array.  Have tested and it is adding germs to the cell it's planted in.

File: ExpandedLights/ExpandedLightsPatch.cs [hunk 6]
```diff
@@ -60,8 +66,8 @@ public static void Prefix()
             }
         }
 
-        [PLibMethod(RunAt.BeforeDbInit)]
-        internal static void DbInitPrefix()
+        [PLibMethod(RunAt.AfterDbInit)]
```
Comment (medkg15) [inline ExpandedLights/ExpandedLightsPatch.cs:69]: Same as above - crashes if this is done in prefix.

File: SkyLib/^SkyLib.csproj [hunk 7]
```diff
@@ -1,23 +1,12 @@
...
-    <Reference Include="Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
-      <HintPath>..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\OxygenNotIncluded\OxygenNotIncluded_Data\Managed\Assembly-CSharp.dll</HintPath>
-    </Reference>
-    <Reference Include="Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed">
-      <HintPath>..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\OxygenNotIncluded\OxygenNotIncluded_Data\Managed\Newtonsoft.Json.dll</HintPath>
-    </Reference>
-    <Reference Include="UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
-      <HintPath>..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\OxygenNotIncluded\OxygenNotIncluded_Data\Managed\UnityEngine.dll</HintPath>
-    </Reference>
-  </ItemGroup>
```
Comment (medkg15) [inline SkyLib/^SkyLib.csproj:22]: Assembly-CSharp and Newtonsoft are handled in .props file, and UnityEngine doesn't appear to be needed.  Maybe some other mods depend on these, though - feel free to add back if needed.

File: DiseasesReimagined/ImaginationLoader.cs (hunk 4 above)
Comment (peterhaneve) [reply DiseasesReimagined/ImaginationLoader.cs:80]: The new merged version of `ImaginationLoader` is available at https://github.com/peterhaneve/ONIMods/blob/main/DecorReimagined/ImaginationLoader.cs.

**Comment by skairunner** [comment]
That's great, thanks for the contribution! If you could just add the new ImaginationLoader that Peter mentioned it seems it'll be good to go

**Comment by medkg15** [comment]
ImaginationLoader has been updated.

**Comment by Crazz2424** [comment]
Is there a built version of the Diseases Restored update I could download to test?

**Comment by skairunner** [comment]
@Crazz2424 I released the updated mods to Steam.

**Comment by Crazz2424** [comment]
Fantastic thank you!

**Pull request merged**